src/iqm/qiskit_iqm/iqm_backend.py

```diff
@@ -92,10 +102,30 @@ def validate_compatible_architecture(self, architecture: QuantumArchitectureSpec
         """Given a quantum architecture specification returns true if its number of qubits, names of qubits and qubit
         connectivity matches the architecture of this backend."""
         qubits_match = set(architecture.qubits) == set(self._quantum_architecture.qubits)
-        ops_match = set(architecture.operations) == set(self._quantum_architecture.operations)
+        ops_match = compare_operations(architecture.operations, self._quantum_architecture.operations)
 
         self_connectivity = set(map(frozenset, self._quantum_architecture.qubit_connectivity))  # type: ignore
         target_connectivity = set(map(frozenset, architecture.qubit_connectivity))  # type: ignore
         connectivity_match = self_connectivity == target_connectivity
 
         return qubits_match and ops_match and connectivity_match
+
+
+def compare_operations(ops1: dict[str, list[list[str]]], ops2: dict[str, list[list[str]]]) -> bool:
+    """Compares the given operation sets defined by the quantum architecture. Returns true if they are the same."""
```

Comment:
Could you clarify `the same` in the docstring to be more precise, e.g. either "equivalent" or "identical".

Reply:
Sure, fixed that